pkg/discovery/discovery.go

```diff
@@ -357,20 +355,38 @@ func updateStatus(client kubernetes.Interface, namespace string, status string,
 	return setPodStatusAnnotation(client, namespace, podStatus)
 }
 
-func updatePluginStatus(client kubernetes.Interface, namespace string, pluginType string, pluginResultStatus string, pluginResultCounts map[string]int) error {
+func updatePluginStatus(client kubernetes.Interface, namespace string, pluginType string, item results.Item) error {
 	podStatus, err := pluginaggregation.GetStatus(client, namespace)
 	if err != nil {
 		return errors.Wrap(err, "failed to get the existing status")
 	}
 
+	integrateResultsIntoStatus(podStatus, pluginType, &item)
+
+	return setPodStatusAnnotation(client, namespace, podStatus)
+}
+
+func integrateResultsIntoStatus(podStatus *pluginaggregation.Status, pluginName string, item *results.Item) {
 	for i := range podStatus.Plugins {
-		if podStatus.Plugins[i].Plugin == pluginType {
-			podStatus.Plugins[i].ResultStatus = pluginResultStatus
-			podStatus.Plugins[i].ResultStatusCounts = pluginResultCounts
-			break
+		if podStatus.Plugins[i].Plugin != pluginName {
+			continue
 		}
+		var itemForNode *results.Item
+		if podStatus.Plugins[i].Node == plugin.GlobalResult {
+			itemForNode = item
+		} else {
+			itemForNode = getItemInTree(item, podStatus.Plugins[i].Node)
```

Comment:
Here I just traverse and choose the item/branch that matches the node name. It works and I think is sufficient, but I also considered adding some sort of metadata field so that we know that not only does `name=NodeName` but it has some `type=hostname` or something like that. Maybe not really necessary though.

Reply:
I think it could help for readability, as I find myself getting lost in what some of these objects are and what they represent. I think just a comment would be enough though to inform readers of the code for next time that the `Name` field for a DS result item will be the node name.

Reply:
Created a new issue; adding it to just nodes doesn't make as much sense as adding it to a few different things (overall summary, file summaries, etc) and so the change is a little more than I want to smush into this PR.